### PR TITLE
Annotate ADAL public API surface with nullability speficiers

### PR DESCRIPTION
**Pull request by oldalton**

### CODE REVIEW

File: ADAL/src/public/ADAuthenticationContext.h [hunk 1]
```diff
@@ -227,56 +227,56 @@ typedef enum
...
+                    response:(nonnull NSURL*)response;
 
 /*!
  */
-+ (BOOL)handleBrokerResponse:(NSURL*)response;
++ (BOOL)handleBrokerResponse:(nonnull NSURL*)response;
 
 /*! Represents the authority used by the context. */
-@property (readonly) NSString* authority;
+@property (readonly, nullable) NSString* authority;
```
Comment (antrix1989) [inline ADAL/src/public/ADAuthenticationContext.h]: should this be nonnull?

File: Samples/SampleSwiftApp/SampleSwiftApp/ViewController.swift [hunk 2]
```diff
@@ -44,40 +44,45 @@ class ViewController: UIViewController {
...
-                                             redirectUri: URL(string: "urn:ietf:wg:oauth:2.0:oob"))
+        guard let authContext = ADAuthenticationContext(authority: "https://login.microsoftonline.com/common", error: nil) else {
+            
+            print("Failed to create auth context")
+            return
+        }
+        
+        authContext.acquireToken(withResource: "https://graph.windows.net",
+                                 clientId: "b92e0ba5-f86e-4411-8e18-6b5f928d968a",
+                                 redirectUri: URL(string: "urn:ietf:wg:oauth:2.0:oob")!)
```
Comment (unpluggedk) [inline Samples/SampleSwiftApp/SampleSwiftApp/ViewController.swift]: self needs to be referenced weak here.
option 1) 
guard self = self else { return } and then use self,
or self? for all the references.

File: Samples/SampleSwiftApp/SampleSwiftApp/ViewController.swift [hunk 3]
```diff
@@ -44,40 +44,45 @@ class ViewController: UIViewController {
...
-                    self.updateStatusField("Unexpected internal error occured");
-                    
-                } else {
+            guard result.status == AD_SUCCEEDED else {
+                
+                if result.error!.domain == ADAuthenticationErrorDomain
+                    && result.error!.code == ADErrorCode.ERROR_UNEXPECTED.rawValue {
                     
-                    self.updateStatusField(result!.error.description)
+                    self.updateStatusField("Unexpected internal error occured")
```
Comment (unpluggedk) [inline Samples/SampleSwiftApp/SampleSwiftApp/ViewController.swift]: check this method calls ui update on the main thread

File: ADAL/src/public/ADAuthenticationContext.h (hunk 1 above)
Comment (oldalton) [reply ADAL/src/public/ADAuthenticationContext.h]: Addressed

File: Samples/SampleSwiftApp/SampleSwiftApp/ViewController.swift (hunk 2 above)
Comment (oldalton) [reply Samples/SampleSwiftApp/SampleSwiftApp/ViewController.swift]: Addressed

File: Samples/SampleSwiftApp/SampleSwiftApp/ViewController.swift (hunk 3 above)
Comment (oldalton) [reply Samples/SampleSwiftApp/SampleSwiftApp/ViewController.swift]: Addressed

**Pull request merged**